src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -349,6 +350,43 @@ public bool IsVisible
             }
         }
 
+        /// <summary>
+        /// Determines whether or not the semi-transparent overlay is displaying on the node.
+        /// This reflects whether the node is in a info/warning/error/frozen state
+        /// </summary>
+        [JsonIgnore]
+        public bool NodeOverlayVisible => IsFrozen ||
+                                          ErrorBubble?.NodeErrorsToDisplay.Count > 0 && WorkspaceViewModel.Zoom < Configurations.ZoomedOutThreshold ||
+                                          ErrorBubble?.NodeWarningsToDisplay.Count > 0 && WorkspaceViewModel.Zoom < Configurations.ZoomedOutThreshold;
+
+        /// <summary>
+        /// The color of the warning bar: blue for info, orange for warnings, red for errors.
+        /// </summary>
+        [JsonIgnore]
+        public SolidColorBrush WarningBarColor
+        {
+            get => warningBarColor;
+            internal set
+            {
+                warningBarColor = value;
+                RaisePropertyChanged(nameof(WarningBarColor));
+            }
+        }
+
+
+        /// <summary>
+        /// Determines the color of the node's visual overlay, which displays
+        /// if the node is in a Frozen, Info, Error or Warning state.
+        /// </summary>
+        [JsonIgnore]
+        public SolidColorBrush NodeOverlayColor
```

Comment:
is this overlay color different in some cases from the warning bar color - seems they could be related to each other to keep the getter a bit more complex but removing some extra state we need to keep in sync?

Reply:
the overlay's a bit more complicated, it needs to take `IsFrozen` into account as well